analyzers/tests/SonarAnalyzer.UnitTest/TestCases/PreferGuidEmpty.Fixed.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System;
+
+class Compliant
+{
+    Guid? nullableField; // Compliant
+    readonly Guid? readonlyNullableField; // Compliant
+    Guid Property { get; set; } // Compliant
+
+    void Empty()
+    {
+        var empty = Guid.Empty; // Compliant
+    }
+
+    void NotEmpty()
+    {
+        var rnd = Guid.NewGuid();// Compliant
+        var bytes = new Guid(new byte[0]); // Compliant
+        var str = new Guid("FA97FFE7-532C-4015-8698-49D8CE4126F4"); // Compliant
+    }
+
+    void NullableDefault()
+    {
+        Guid? nullable = default; // Compliant, not equivalent to Guid.Empty.
+        var nullableOfT = default(Guid?); // Compliant
+        var guidAsParameter = new NullableGuidClass(default); // Compliant
+        guidAsParameter.Method(default); // Compliant
+    }
+
+    void NotInitialized(string str)
+    {
+        Guid parsed; // Compliant
+        Guid.TryParse(str, out parsed);
+    }
+
+    void OptionalParameter(Guid id = default) { } // Compliant, default has to be a run-time constant
+
+    void NotGuid()
+    {
+        int integer = default;
+        var date = default(DateTime);
+    }
+}
+
+class NonCompliant
+{
+    Guid field; // FN
+    readonly Guid readonlyField; // FN
+    Guid Property { get; set; }
+
+    void DefaultCtor()
+    {
+        var ctor = Guid.Empty; // Fixed
+    }
+
+    void DefaultInintiation()
+    {
+        Guid defaultValue = Guid.Empty; // Fixed
+        var defaultOfGuid = Guid.Empty; // Fixed
+        Property = Guid.Empty; // Fixed
+        var instance = new GuidClass(Guid.Empty); // Fixed
+        instance.Method(Guid.Empty); // Fixed
+    }
+
+    void EmptyString()
+    {
+        var emptyCtor = Guid.Empty; // Fixed
+        var emptyParse = Guid.Parse("00000000-0000-0000-0000-000000000000"); // FN
+    }
+}
+
+struct GuidAssignmentStruct
+{
+    static readonly Guid Static; // FN
```

Comment:
I would not mark it as FN
```suggestion
    static readonly Guid Static; 
```

Reply:
Well, that was my question (earlier): do we consider this a place where `Guid.Empty` is preferred. I would argue it does, but is a totally different approach to detect it. If we decide that we do not want to report on this, that's fine too.

Reply:
I'm answering that question here :) I'm afraid we would have way too many FPs if we'd try to detect this scenario